### PR TITLE
Add support for a second bazel version in kubekins-e2e

### PR DESCRIPTION
**Pull request by fejta**
/assign @mikedanese @Katharine

### CODE REVIEW

File: images/kubekins-e2e/Dockerfile [hunk 1]
```diff
@@ -74,6 +76,9 @@ ENV BAZEL_VERSION=${BAZEL_VERSION_ARG}
 COPY ./install-bazel.sh /
 RUN bash /install-bazel.sh
 
+COPY --from=old \
+    /usr/local/lib/bazel/bin/bazel-real /usr/local/lib/bazel/bin/bazel-${OLD_VERSION}
```
Comment (mikedanese) [inline images/kubekins-e2e/Dockerfile]:
```suggestion
    /usr/local/lib/bazel/bin/bazel-real /usr/local/lib/bazel/bin/bazel-${OLD_BAZEL_VERSION?}
```

Comment (fejta) [reply images/kubekins-e2e/Dockerfile]: Done

File: images/kubekins-e2e/Dockerfile [hunk 2]
```diff
@@ -74,6 +76,9 @@ ENV BAZEL_VERSION=${BAZEL_VERSION_ARG}
 COPY ./install-bazel.sh /
 RUN bash /install-bazel.sh
 
+COPY --from=old \
```
Comment (mikedanese) [inline images/kubekins-e2e/Dockerfile]: Add
```
ARG OLD_BAZEL_VERSION
```
Again before this line. Previous declaration was not in this stage.

Comment (fejta) [reply images/kubekins-e2e/Dockerfile]: Done

**Comment by justaugustus** [comment]
@fejta @mikedanese -- so... how do we use it?

**Comment by fejta** [comment]
$REPOROOT/.bazelversion determines the actual version of bazel used.

Set it to `0.23.2` and it will run with `0.23.2`
Set it to `2.2.0` and it will run with `2.2.0`
Set it to `2.1.0` and it will error with a message of how to install that version (I only setup the image to come with two versions preinstalled).

This is the default behavior of bazel 1.0+ (and the default behavior of [bazelisk](https://github.com/bazelbuild/bazelisk))

**Comment by BenTheElder** [comment]
trying to figure out which change caused breakage, not long after this the 1.15 bazel e2e build started failing, one of the only changes is the kubekins image bump https://testgrid.k8s.io/conformance-all#kind,%20v1.15%20%28dev%29

**Comment by BenTheElder** [comment]
we don't have a .bazelversion on branches prior to 1.17

**Comment by fejta** [comment]
Nice find

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/kubernetes/pull/89446 & https://github.com/kubernetes/kubernetes/pull/89448

**Comment by justaugustus** [comment]
Both CPs have been approved. Thanks y'all!!

**Comment by mikedanese** [comment]
ref https://github.com/kubernetes/kubernetes/issues/89414

**Pull request merged**